src/ir/info.rs

```diff
@@ -1,76 +1,79 @@
 use super::{Component, DisplayCtx, ExprIdx, Range, TimeIdx, TimeSub};
 use crate::{ast, utils::GPosIdx};
 use codespan_reporting::diagnostic::Diagnostic;
+use struct_variant::struct_variant;
+
+pub struct Empty;
+
+pub struct Assert(pub Reason);
+
+pub struct Param {
+    pub name: ast::Id,
+    pub bind_loc: GPosIdx,
+}
+
+pub struct Event {
+    pub name: ast::Id,
+    pub bind_loc: GPosIdx,
+    pub delay_loc: GPosIdx,
+    pub interface_name: Option<ast::Id>,
+    pub interface_bind_loc: Option<GPosIdx>,
+}
+
+pub struct EventBind {
+    pub ev_delay_loc: GPosIdx,
+    pub bind_loc: GPosIdx,
+}
+
+pub struct Instance {
+    pub name: ast::Id,
+    pub comp_loc: GPosIdx,
+    pub bind_loc: GPosIdx,
+}
+
+pub struct Invoke {
+    pub name: ast::Id,
+    pub inst_loc: GPosIdx,
+    pub bind_loc: GPosIdx,
+}
+
+pub struct Connect {
+    pub dst_loc: GPosIdx,
+    pub src_loc: GPosIdx,
+}
+
+pub struct Port {
+    pub name: ast::Id,
+    pub bind_loc: GPosIdx,
+    pub width_loc: GPosIdx,
+    pub live_loc: GPosIdx,
+}
```

Comment:
This removed all the documentation information from the fields. We should try to preserve those.